### PR TITLE
add verify-golangci-lint.sh script

### PR DESCRIPTION
**Pull request by rlankfo**
Started work on https://github.com/kubernetes/kops/issues/12821 here, and have added a `verify-golangci-lint.sh` script which can be run by hand.

This doesn't replace staticcheck yet but I wanted to get some initial feedback on the script before making further updates.

/cc @rifelpet

### CODE REVIEW

**Comment by rlankfo** [comment]
I signed it

**Comment by rlankfo** [comment]
@rifelpet did you have a chance to review this? Can you give me some guidance on next steps forward? Thanks.

**Comment by rifelpet** [comment]
@rlankfo I've added a new presubmit job in https://github.com/kubernetes/test-infra/pull/24651.

You'll need to add a new make target similar to [staticcheck's target](https://github.com/kubernetes/kops/blob/d0b577183fe9df0f562905d1ff9190262636b38b/Makefile#L420). Once that is added we can test it out with `/test pull-kops-verify-golangci-lint`

**Comment by rlankfo** [comment]
thanks @rifelpet 🙏 

Looks like it's running: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kops/12892/pull-kops-verify-golangci-lint/1470175375601438720

File: .golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+linters-settings: # please keep this alphabetized
+  staticcheck:
+    go: "1.17"
+    checks: [
+      "all",
+      "-S1*",    # TODO(fix) Omit code simplifications for now.
+      "-ST1*",   # Mostly stylistic, redundant w/ golint
+      "-SA5011", # TODO(fix) Possible nil pointer dereference
+      "-SA1019", # TODO(fix) Using a deprecated function, variable, constant or field
+      "-SA2002"  # TODO(fix) Called testing.T.FailNow or SkipNow in a goroutine, which isn’t allowed
```
Comment (rifelpet) [inline .golangci.yaml:36]:
```suggestion
      "-ST1000",  # Incorrect or missing package comment
      "-ST1003",  # Poorly chosen identifier
      "-ST1005",  # Incorrectly formatted error string
      "-ST1006",  # Poorly chosen receiver name
      "-ST1012",  # Poorly chosen name for error variable
      "-ST1016",  # Use consistent method receiver names
      "-ST1020",  # Comment on exported method should be of the form ...
      "-ST1021",  # Comment on exported type should be of the form ...
      "-ST1022",  # Comment on exported const should be of the form ...
      "-SA5011"   # TODO(fix) Possible nil pointer dereference
```

Pulling these from the existing hack/verify-staticcheck.sh:

https://github.com/kubernetes/kops/blob/88a9d50e7321b185c675540f7035dd203e450a44/hack/verify-staticcheck.sh#L30-L42

**Review by rifelpet — APPROVED** [review]

**Pull request merged**